msteams-platform/samples/virtual-assistant.md

```diff
@@ -0,0 +1,560 @@
+    {
+        try
+        {
+            string skillId = (turnContext.Activity as Activity).GetSkillId();
+            var skill = _skillsConfig.Skills.Where(s => s.Value.AppId == skillId).FirstOrDefault().Value;
+
+            // Forward request to correct skill
+            var invokeResponse = await _skillHttpClient.PostActivityAsync(this._appId, skill, _skillsConfig.SkillHostEndpoint, turnContext.Activity as Activity, cancellationToken);
+
+            return invokeResponse.GetTaskModuleRespose();
+        }
+        catch (Exception exception)
+        {
+            await turnContext.SendActivityAsync(_templateEngine.GenerateActivityForLocale("ErrorMessage"));
+            _telemetryClient.TrackException(exception);
+
+            return null;
+        }
+    }
+
+    // Invoked when a 'task/submit' invoke activity is received for task module submit actions.
+    protected override async Task<TaskModuleResponse> OnTeamsTaskModuleSubmitAsync(ITurnContext<IInvokeActivity> turnContext, TaskModuleRequest taskModuleRequest, CancellationToken cancellationToken)
+    {
+        try
+        {
+            string skillId = (turnContext.Activity as Activity).GetSkillId();
+            var skill = _skillsConfig.Skills.Where(s => s.Value.AppId == skillId).FirstOrDefault().Value;
+
+            // Forward request to correct skill
+            var invokeResponse = await _skillHttpClient.PostActivityAsync(this._appId, skill, _skillsConfig.SkillHostEndpoint, turnContext.Activity as Activity, cancellationToken).ConfigureAwait(false);
+
+            return invokeResponse.GetTaskModuleRespose();
+        }
+        catch (Exception exception)
+        {
+            await turnContext.SendActivityAsync(_templateEngine.GenerateActivityForLocale("ErrorMessage"));
+            _telemetryClient.TrackException(exception);
+
+            return null;
+        }
+    }
+```
+
+Additionally, all skill domains must be included in the `validDomains` section in Virtual Assistant's manifest file so that task modules invoked via a skill render properly.
+
+### Handling collaborative app scopes
+
+Teams apps can exist in multiple scopes including 1:1 chat, group chat, and channels. The core Virtual Assistant template is designed for 1:1 chats. As part of the onboarding experience Virtual Assistant  prompts users for name and maintains user state. Since that onboarding experience is not suited for group chat/channel scopes it has been removed.
+
+Skills should handle activities in multiple scopes (1:1 chat, group chat, and channel conversation). If any of these scopes are not supported, skills should respond with an appropriate message.
+
+The following  processing functions have been added to Virtual Assistant core:
+
+- Virtual Assistant can be invoked without any text message from a group chat or channel.
+- Articulations are cleaned (i.e.,  remove @mentions) prior to it being sent to the dispatch module so that the dispatch output is unaffected.
```

Comment:
Let's just say here that we remove the necessary at-mention of the bot before sending the message to the dispatch module.